server/mailet/integration-testing/src/test/java/org/apache/james/mailets/RemoteDeliveryErrorTest.java

```diff
@@ -370,6 +382,57 @@ public void remoteDeliveryShouldNotDuplicateContentWhenSendPartial() throws Exce
                 "]");
     }
 
+    @Ignore("JAMES-2097 Using full recipients for following MX iteration when partial fails on delivering")
+    @Test
+    public void remoteDeliveryShouldNotDuplicateContentWhenSendPartialWhenFailover() throws Exception {
+        ImmutableList<InetAddress> addresses = ImmutableList.of(InetAddress.getByName(mockSmtp.getContainerIp()));
+        ImmutableList<String> mxs = ImmutableList.of(mockSmtp.getContainerIp(), mockSmtp2.getContainerIp());
+        ImmutableList<String> txtRecords = ImmutableList.of();
+
+        inMemoryDNSService.registerRecord(ANOTHER_DOMAIN, addresses, mxs, txtRecords)
+            .registerMxRecord(mockSmtp.getContainerIp(), mockSmtp.getContainerIp())
+            .registerMxRecord(mockSmtp2.getContainerIp(), mockSmtp2.getContainerIp());
+
+        with()
+            .body(ALWAYS_PARTIAL_RCPT_421_BEHAVIOR)
+            .put("/smtpBehaviors");
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .sendMessage(MailImpl.builder()
+                .name("name")
+                .sender(new MailAddress(FROM))
+                .addRecipient(RECIPIENT1)
+                .addRecipient(RECIPIENT2)
+                .mimeMessage(MimeMessageUtil.mimeMessageFromString(MIME_MESSAGE))
+                .build());
+
+        awaitAtMostOneMinute.untilAsserted(() -> given()
+            .get("/smtpMails")
+        .then()
+            .body("", hasSize(1)));
+        awaitAtMostOneMinute.untilAsserted(() -> given(requestSpecification(mockSmtp2.getContainerIp()), RESPONSE_SPECIFICATION)
+            .get("/smtpMails")
+        .then()
+            .body("", hasSize(1)));
+
+        given()
+            .get("/smtpMails")
+        .then()
+            .body("", hasSize(1))
+            .body("[0].from", is(FROM))
+            .body("[0].recipients", hasSize(1))
+            .body("[0].recipients[0]", is(RECIPIENT2))
+            .body("[0].message", containsString("subject: test"));
+        given(requestSpecification(mockSmtp2.getContainerIp()), RESPONSE_SPECIFICATION)
+            .get("/smtpMails")
+        .then()
+            .body("", hasSize(1))
+            .body("[0].from", is(FROM))
+            .body("[0].recipients", hasSize(1))
+            .body("[0].recipients[0]", is(RECIPIENT1))
+            .body("[0].message", containsString("subject: test"));
```

Comment:
I can contribute a convenience MockSMTPClient with Feign

Reply:
Let's keep it simple?